src/main/java/com/exasol/report/ReportWriter.java

```diff
@@ -0,0 +1,98 @@
+package com.exasol.report;
+
+import java.io.*;
+import java.nio.file.Path;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.List;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
+
+import com.exasol.UserInput;
+
+/**
+ * This class writes reports from {@link ValidationReport} to a file.
+ */
+public class ReportWriter {
+    private static final Logger LOGGER = Logger.getLogger(ReportWriter.class.getName());
+    private static final String LINE_SEPARATOR = System.getProperty("line.separator");
+    private final UserInput userInput;
+    private final Path reportPath;
+
+    /**
+     * Create a new instance of {@link ReportWriter}.
+     *
+     * @param userInput instance of {@link UserInput}
+     */
+    public ReportWriter(final UserInput userInput, final Path reportPath) {
+        this.userInput = userInput;
+        this.reportPath = reportPath;
+    }
+
+    /**
+     * Write a validation report to the file.
+     *
+     * @param reports one or more reports to write
+     */
+    // [impl->dsn~rr-writes-validation-report-to-file~1]
+    public void writeValidationReportToFile(final List<Report> reports) {
+        final File reportFile = prepareFile();
+        try (final FileWriter writer = new FileWriter(reportFile.getAbsoluteFile())) {
+            writeExecutionInformation(writer);
+            writeReports(writer, reports);
+        } catch (final IOException exception) {
+            throw new IllegalStateException("E-RR-RW-2: Unable to write a report.", exception);
+        }
+        LOGGER.info(() -> "A full report is available: " + this.reportPath.toString());
```

Comment:
move into try

Reply:
I think it's better when a try block contains as few lines as possible. There is no need to move it there